app/views/carts/show.html.erb

```diff
@@ -39,7 +39,7 @@
     </div>
     <div>
       <div class="col-xs-2">
-        <%= button_to "Checkout", profile_orders_path, class: "btn btn-success btn-block" %>
+        <%= button_to "Checkout", profile_orders_path, class: "btn btn-success btn-block" if current_user && current_user.registered? && session[:cart] != nil && session[:cart] != {} %>
```

Comment:
We might be able to just use `@cart.contents.empty?` instead of the extra booleans as it would always be initialized before viewing the page

Reply:
Good idea! I just implemented it and it still works.